src/meta/src/telemetry.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::SystemTime;
+
+use anyhow::anyhow;
+use risingwave_common::telemetry::{
+    post_telemetry_report, telemetry_env_enabled, SystemData, TelemetryNodeType,
+    TelemetryReportBase, TELEMETRY_REPORT_INTERVAL, TELEMETRY_REPORT_URL,
+};
+use serde::{Deserialize, Serialize};
+use tokio::sync::oneshot::Sender;
+use tokio::task::JoinHandle;
+use tokio::time::{interval, Duration};
+use uuid::Uuid;
+
+use crate::storage::MetaStore;
+
+pub const TELEMETRY_CF: &str = "cf/telemetry";
+/// `telemetry` in bytes
+pub const TELEMETRY_KEY: &[u8] = &[74, 65, 0x6c, 65, 0x6d, 65, 74, 72, 79];
+
+#[derive(Debug, Serialize, Deserialize)]
+struct TelemetryReport {
+    #[serde(flatten)]
+    base: TelemetryReportBase,
+}
+
+/// This function spawns a new tokio task to report telemetry.
+/// It creates a channel for killing itself and a join handle to the spawned task.
+/// It then creates an interval of `TELEMETRY_REPORT_INTERVAL` seconds and checks if telemetry is
+/// enabled. If it is, it gets or creates a tracking ID from the meta store,
+/// creates a `TelemetryReport` object with system data, uptime, timestamp, and tracking ID.
+/// Finally, it posts the report to `TELEMETRY_REPORT_URL`.
+/// If an error occurs at any point in the process, it logs an error message.
+pub async fn start_meta_telemetry_reporting(
```

Comment:
`start_meta_telemetry_reporting` seems to be very similar to `start_telemetry_reporting`. Wonder whether we can resue some common logics to avoid code duplication.